fastapi/middleware.py

```diff
@@ -13,12 +13,26 @@
 
 import a2wsgi
 from a2wsgi.asgi import ASGIResponder
+from a2wsgi.asgi_typing import ASGIApp
 from a2wsgi.wsgi_typing import Environ, StartResponse
 
 from .pools import event_loop_pool
 
 
 class ASGIMiddleware(a2wsgi.ASGIMiddleware):
+    def __init__(
+        self,
+        app: ASGIApp,
+        wait_time: float | None = None,
```

Comment:
Shouldn't we need to keep the parent signature?

`loop: Optional[asyncio.AbstractEventLoop] = None,`

Reply:
IMO no. This class is designed to be used in the specific context of the fastapi addon where we enforce the use of the pool of eventloop threads.